src/control/lib/daos/pool.go

```diff
@@ -0,0 +1,246 @@
+		Label            string               `json:"label,omitempty"`
+		TotalTargets     uint32               `json:"total_targets"`
+		ActiveTargets    uint32               `json:"active_targets"`
+		TotalEngines     uint32               `json:"total_engines"`
+		DisabledTargets  uint32               `json:"disabled_targets"`
+		Version          uint32               `json:"version"`
+		Leader           uint32               `json:"leader"`
+		ServiceReplicas  []ranklist.Rank      `json:"svc_reps"`
+		Rebuild          *PoolRebuildStatus   `json:"rebuild"`
+		TierStats        []*StorageUsageStats `json:"tier_stats"`
+		EnabledRanks     *ranklist.RankSet    `json:"-"`
+		DisabledRanks    *ranklist.RankSet    `json:"-"`
+		PoolLayoutVer    uint32               `json:"pool_layout_ver"`
+		UpgradeLayoutVer uint32               `json:"upgrade_layout_ver"`
+	}
+
+	// StorageTargetUsage represents DAOS target storage usage
+	StorageTargetUsage struct {
+		Total     uint64           `json:"total"`
+		Free      uint64           `json:"free"`
+		MediaType StorageMediaType `json:"media_type"`
+	}
+)
+
+func (srs *StorageUsageStats) calcImbalance(targCount uint32) uint32 {
+	spread := srs.Max - srs.Min
+	return uint32((float64(spread) / (float64(srs.Total) / float64(targCount))) * 100)
+}
+
+func (pi *PoolInfo) Usage() []*PoolTierUsage {
+	var tiers []*PoolTierUsage
+	for _, tier := range pi.TierStats {
+		tiers = append(tiers, &PoolTierUsage{
+			TierName:  strings.ToUpper(tier.MediaType.String()),
+			Size:      tier.Total,
+			Free:      tier.Free,
+			Imbalance: tier.calcImbalance(pi.ActiveTargets),
+		})
+	}
+	return tiers
+}
+
+// RebuildState returns a string representation of the pool rebuild state.
+func (pi *PoolInfo) RebuildState() string {
+	if pi.Rebuild == nil {
+		return "Unknown"
+	}
+	return pi.Rebuild.State.String()
+}
+
+// Name retrieves effective name for pool from either label or UUID.
+func (pi *PoolInfo) Name() string {
+	name := pi.Label
+	if name == "" {
+		// use short version of uuid if no label
+		name = strings.Split(pi.UUID.String(), "-")[0]
+	}
+	return name
+}
+
+// PoolServiceState is used to represent the state of the pool service
+type PoolServiceState uint
+
+const (
+	// PoolServiceStateCreating indicates that the pool service is being created
+	PoolServiceStateCreating PoolServiceState = iota
+	// PoolServiceStateReady indicates that the pool service is ready to be used
+	PoolServiceStateReady
+	// PoolServiceStateDestroying indicates that the pool service is being destroyed
+	PoolServiceStateDestroying
+	// PoolServiceStateDegraded indicates that the pool service is in a degraded state
+	PoolServiceStateDegraded
+	// PoolServiceStateUnknown indicates that the pool service state is unknown
+	PoolServiceStateUnknown
+)
+
+func (pss PoolServiceState) String() string {
+	if pss < PoolServiceStateCreating || pss > PoolServiceStateUnknown {
+		return fmt.Sprintf("Invalid pool service state %d", pss)
+	}
```

Comment:
As I was looking at these stringers and marshallers, I realized that we were in an in-between state, where some of the types were backed by the protobuf values, and some were just using native values. I started to go down the road of getting rid of all of the protobuf stuff, with the thought that the `lib/daos` package shouldn't depend on the protobufs. I backed away from that position though, because I realized that I was adding maintenance burden to keeping everything in sync.

I still think there's value in revisiting whether or not we should have protobuf dependencies in `lib/daos`, but I don't think this PR is the place to do that.